src/app/pages/login/login-form/login-form.component.ts

```diff
@@ -0,0 +1,69 @@
+import { Component, OnInit } from '@angular/core';
+import { Router } from '@angular/router';
+import {FormBuilder, FormGroup, Validators} from '@angular/forms';
+import { LoginService } from '../../../services/login/login.service';
+import { LoginData } from '../../../models/login/loginData';
+import { ToastrService } from 'ngx-toastr';
+
+@Component({
+  selector: 'app-login-form',
+  templateUrl: './login-form.component.html',
+  styleUrls: ['./login-form.component.scss']
+})
+export class LoginFormComponent implements OnInit {
+  loginForm: FormGroup;
+  submitted: boolean = false;
+  success: boolean = false;
+  inputError: boolean = false;
+  
+  constructor(
+    private formBuilder: FormBuilder,
+    private loginService: LoginService,
+    private router: Router,
+    private toastrService: ToastrService,
+    ){}
+
+
+  notification: string ='Provide your email and password';
+  email: boolean = true
+  password: boolean = true;
+  
+  onLogin(loginData: LoginData) {
+    this.submitted = true;
+    if(this.loginForm.invalid){
+      this.notification = 'Error, ensure you provide valid details';
+      this.setErrorTimeout()
+      return;
+    }
+    this.success = true;
+
+    this.loginService.login(loginData).subscribe(response => {
+      this.toastrService.success(response.data.message);
+      this.notification = 'Login was succesful';
+      localStorage.setItem('token', response.data.user.token)
+
+      this.router.navigate([''])
```

Comment:
@BrianSerem, I believe you might need to take advantage of router guards to prevent authenticated users from accessing the login component when they are logged in. As is now, after a successful login a backward navigation will still show users the login component.

Reply:
I believe we earlier spoke about this. We need a new story to handle guarding the routes we currently have. Will gladly take it up after I am done with this feature.

Reply:
great @BrianSerem